code/modules/reagents/chemistry/reagents/food.dm

```diff
@@ -354,7 +354,8 @@
 	var/update_flags = STATUS_UPDATE_NONE
 	if(ishuman(M))
 		var/mob/living/carbon/human/H = M
-		if(H.mind && H.mind.vampire && !H.mind.vampire.get_ability(/datum/vampire_passive/full)) //incapacitating but not lethal.
+		var/datum/antagonist/vampire/V = H.mind.has_antag_datum(/datum/antagonist/vampire)
```

Comment:
You need a safety check here, because you could trigger this on mindless humans and make runtimes.
```suggestion
		var/datum/antagonist/vampire/V = H.mind?.has_antag_datum(/datum/antagonist/vampire)
```